cartridges/int_adyen_SFRA/cartridge/client/default/js/adyenCheckout.js

```diff
@@ -165,9 +169,101 @@
+                var ssnContainer = document.createElement("div");
+                $(ssnContainer).attr('id', 'ssn_' + paymentMethod.type);
+                var socialSecurityNumberLabel = document.createElement("span");
+                $(socialSecurityNumberLabel).text("Social Security Number").attr('class', 'adyen-checkout__label');
+                var socialSecurityNumber = document.createElement("input");
+                $(socialSecurityNumber).attr('id', 'ssnValue').attr('class', 'adyen-checkout__input').attr('type', 'text').attr('maxlength', ssnLength);
+
+                ssnContainer.append(socialSecurityNumberLabel);
+                ssnContainer.append(socialSecurityNumber);
+                klarnaContainer.append(ssnContainer);
+            }
+
+            li.append(klarnaContainer);
+
+        };
+
+        if(paymentMethod.type.indexOf("afterpay_default") !== -1){
+            var afterpayContainer = document.createElement("div");
+            $(afterpayContainer).addClass('hppAdditionalFields').attr('id', 'component_' + paymentMethod.type).attr('style', 'display:none');
+            afterpayComponent = checkout.create('afterpay', {
+                countryCode: $('#currentLocale').val(),
+                details: filterOutOpenInvoiceComponentDetails(paymentMethod.details),
+                visibility: {
+                    personalDetails: "editable"
+                }
+            });
+            li.append(afterpayContainer);
+            afterpayComponent.mount(afterpayContainer);
+        };
+
         $('#paymentMethodsUl').append(li);
     };
 
+
+    //Filter fields for open invoice validation
+    function filterOutOpenInvoiceComponentDetails(details) {
+        var filteredDetails = details.map(function (detail) {
+            if (detail.key == "personalDetails") {
+                var detailObject = detail.details.map(function (childDetail) {
+                    if (childDetail.key == 'dateOfBirth' ||
+                        childDetail.key == 'gender') {
+                        return childDetail;
+                    }
+                });
+
+                if (!!detailObject) {
+                    return {
+                        "key": detail.key,
+                        "type": detail.type,
+                        "details": filterUndefinedItemsInArray(detailObject)
+                    };
+                }
+            }
+        });
+
+        return filterUndefinedItemsInArray(filteredDetails);
+    };
+
+    /**
+     * Helper function to filter out the undefined items from an array
+     * @param arr
+     * @returns {*}
+     */
+    function filterUndefinedItemsInArray(arr) {
+        return arr.filter(function (item) {
+            return typeof item !== 'undefined';
+        });
+    };
+
+    function getSsnLengthNordicCountry(country){
+        if(country.indexOf("NO") > -1 || country.indexOf("SE") > -1 || country.indexOf("FI") > -1 || country.indexOf("DK") > -1){
```

Comment:
Do we need to check it with the indexof? I mean we can simply check if 

```suggestion
        if(country === "NO" || country === "SE"|| country === "FI" || country === "DK"){
```